rulette-engine/src/test/java/com/github/kislayverma/rulette/engine/impl/trie/TrieBasedEvaluationEngineTest.java

```diff
@@ -0,0 +1,68 @@
+package com.github.kislayverma.rulette.engine.impl.trie;
+
+import com.github.kislayverma.rulette.core.exception.RuleConflictException;
+import com.github.kislayverma.rulette.core.gaia.RuleMother;
+import com.github.kislayverma.rulette.core.gaia.RuleSystemMetaDataMother;
+import com.github.kislayverma.rulette.core.metadata.RuleSystemMetaData;
+import com.github.kislayverma.rulette.core.rule.Rule;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Unit Test class for TrieBasedEvaluationEngine.
+ * @author arpit.jain
+ */
+public class TrieBasedEvaluationEngineTest {
+
+    private static RuleSystemMetaData metaData;
+    private static TrieBasedEvaluationEngine sut;
+
+    @BeforeAll
+    static void beforeAll() {
+        try {
+            metaData = RuleSystemMetaDataMother.getDefaultMetaData();
+        } catch (Exception ex) {
+            fail("Exception while getting default rule system meta data", ex);
+        }
+    }
+
+    @Nested
+    @DisplayName("addRule")
+    class AddRule {
+
+        @BeforeEach
+        void setUp() {
+            try {
+                // Refresh rule engine before each test
+                sut = new TrieBasedEvaluationEngine(metaData);
+            } catch (Exception ex) {
```

Comment:
We shouldnt have to catch exceptions anymoe

Reply:
@kislayverma Just rebased with upstream. The constructor still throws `RuleConflictException`. We will still need to catch that. Are you planning to somehow remove throwing this from the constructor?

Reply:
No I meant that we should only catch the specific exception instead of generic "Exception"

Reply:
No I meant we should catch "RuleConflictException" rather than "Exception"